src/components/ParamsBox.tsx

```diff
@@ -39,19 +39,24 @@ const ParamsBox: FunctionComponent<{}> = () => {
 		selectedEntity.params.setInfiniteType(ev.target.value as AmmoType);
 	}
 
-	const pattern = '^(?:b|g)*$';
-
 	function onChangeMagazine(ev: ChangeEvent<HTMLInputElement>): void {
 		if (selectedEntity === undefined) return;
 		if (!Hoppi.is(selectedEntity)) throw new Error('Not a hoppi');
 		if (!FiniteParams.is(selectedEntity.params)) throw new Error('Params aren\'t finite');
 
-		if (!new RegExp(pattern).test(ev.target.value)) {
+		if (!/^(?:b|g)+$/.test(ev.target.value)) {
 			alert('The content of the magazine must be a string of \'g\' for Grenades and \'b\' for Bombs.');
 		} else {
 			selectedEntity.params.setFromStringFormat(ev.target.value);
 		}
-	}
+       }
+       function onChangeText(ev: ChangeEvent<HTMLInputElement>): void {
+                if (selectedEntity === undefined) return;
+	        if (!Text.is(selectedEntity)) throw new Error('Not a Text'); //typing this by hand makes me wish i copypasted the mag code.
+	        else { 
```

Comment:
You can remove the `else` here and use the early return style.

Since the previous cases return or throw, the return is unnecessary and the code is clearer without it. :slightly_smiling_face:

Reply:
Fixed